src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -0,0 +1,191 @@
+  googleBucketLocation: string // historically returns defaultLocation if cannot be retrieved or Azure
+  googleBucketType: string // historically returns locationTypes.default if cannot be retrieved or Azure
+  azureContainerRegion: string|undefined
+  azureContainerUrl: string|undefined
+  azureContainerSasUrl: string|undefined
+}
+
+export type InitializedWorkspaceWrapper = WorkspaceWrapper & { workspaceInitialized: boolean }
+
+interface WorkspaceDetails {
+  workspace: InitializedWorkspaceWrapper
+  accessError: boolean
+  loadingWorkspace: boolean
+  storageDetails: StorageDetails
+  refreshWorkspace: () => {}
+}
+
+export const GOOGLE_PERMISSIONS_RECHECK_RATE = 15000
+export const AZURE_BUCKET_RECHECK_RATE = 5000
+
+export const useWorkspace = (namespace, name) : WorkspaceDetails => {
+  const [accessError, setAccessError] = useState(false)
+  const [loadingWorkspace, setLoadingWorkspace] = useState(false)
+  const accessNotificationId = useRef()
+  const cachedWorkspace = useStore(workspaceStore)
+  const workspace = cachedWorkspace && _.isEqual({ namespace, name }, _.pick(['namespace', 'name'], cachedWorkspace.workspace)) ? cachedWorkspace : undefined
+  const [{ location, locationType }, setGoogleStorage] = useState({
+    location: defaultLocation, locationType: locationTypes.default // These default types are historical
+  })
+  const [azureStorage, setAzureStorage] = useState<{ location: string; storageContainerUrl: string | undefined; sasUrl: string }>()
+  const workspaceInitialized = workspace?.workspaceInitialized // will be stored in cached workspace
+
+  const signal = useCancellation()
+  const checkInitializationTimeout = useRef<any>()
+
+  const updateWorkspaceInStore = (workspace, initialized) => {
+    workspace.workspaceInitialized = initialized
+    // clone the workspace to force React to re-render components that depend on workspace
+    workspaceStore.set(_.clone(workspace))
+  }
+
+  const checkWorkspaceInitialization = async workspace => {
+    if (!!workspace && !workspaceInitialized) {
+      if (isGoogleWorkspace(workspace)) {
+        await checkGooglePermissions(workspace)
+      } else {
+        await checkAzureStorageExists(workspace)
+      }
+    } else if (!!workspace && isGoogleWorkspace(workspace)) {
+      // console.log('Google, skipping permissions initialization check')
+      await loadGoogleBucketLocation(workspace)
+    } else if (!!workspace && isAzureWorkspace(workspace)) {
+      // console.log('Azure, skipping storage initialization check')
+      await loadAzureStorageDetails(workspace)
+    }
+  }
+
+  const checkGooglePermissions = async workspace => {
+    try {
+      await Ajax(signal).Workspaces.workspace(namespace, name).checkBucketReadAccess()
+      // console.log('got success status!!!!!!')
+      updateWorkspaceInStore(workspace, true)
+      loadGoogleBucketLocation(workspace)
+    } catch (error) {
+      // @ts-ignore
+      const errorText = await error.text()
```

Comment:
@msilva-broad I don't think I have any options here-- do you agree?

Reply:
uresolved comment just to clarify for folks who may see this PR.
- using any as the error type is reflecting reality, we do a lot of unsafe error handling stuff like this where errors may or may not be regular Error types.... sometimes it's a Fetch response object instead, or something else.... Improving error handling and type safety around it is a known gap on my list.  Watch for a PR from @n8landolt that makes an incremental improvement on this front.